python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,429 @@
+
+        # specify file
+        fsurf_in = os.path.join(indir, file)
+        fsurf_out = add_tag_to_filename(fsurf_in, self.tag)
+        logger.info("fsurf_in:  %s", fsurf_in)
+        logger.info("fsurf_out: %s", os.path.join(self.out_dir, fsurf_out))
+
+        # create 1d coordinate variables to enable sel() method
+        f_in = self.create_1d_coord(fsurf_in, "LONGXY", "LATIXY", "lsmlon", "lsmlat")
+
+        # extract gridcell closest to plon/plat
+        f_out = f_in.sel(lsmlon=self.plon, lsmlat=self.plat, method="nearest")
+
+        # expand dimensions
+        f_out = f_out.expand_dims(["lsmlat", "lsmlon"]).copy(deep=True)
+
+
+        # modify surface data properties
+        if self.overwrite_single_pft:
+            f_out["PCT_NAT_PFT"][:, :, :] = 0
+            if self.dom_pft < 16:
+                f_out['PCT_NAT_PFT'][:, :, self.dom_pft] = 100
+        if not self.include_nonveg:
+            f_out["PCT_NATVEG"][:, :] = 100
+            f_out["PCT_CROP"][:, :] = 0
+            f_out["PCT_LAKE"][:, :] = 0.0
+            f_out["PCT_WETLAND"][:, :] = 0.0
+            f_out["PCT_URBAN"][:, :, ] = 0.0
+            f_out["PCT_GLACIER"][:, :] = 0.0
+        if self.uni_snow:
+            f_out["STD_ELEV"][:, :] = 20.0
+        if self.cap_saturation:
+            f_out["FMAX"][:, :] = 0.0
+
+        # specify dimension order
+        f_out = f_out.transpose(
+            u"time",
+            u"cft",
+            u"lsmpft",
+            u"natpft",
+            u"nglcec",
+            u"nglcecp1",
+            u"nlevsoi",
+            u"nlevurb",
+            u"numrad",
+            u"numurbl",
+            "lsmlat",
+            "lsmlon",
+        )
+
+        # update lsmlat and lsmlon to match site specific instead of the nearest point
+        # we do this so that if we create user_mods the PTS_LON and PTS_LAT in CIME match
+        # the surface data coordinates - which is required
+        f_out['lsmlon'] = np.atleast_1d(self.plon)
+        f_out['lsmlat'] = np.atleast_1d(self.plat)
+        f_out['LATIXY'][:, :] = self.plat
+        f_out['LONGXY'][:, :] = self.plon
+
+        # update attributes
+        self.update_metadata(f_out)
+        f_out.attrs["Created_from"] = fsurf_in
+        # mode 'w' overwrites file
+        wfile = os.path.join(self.out_dir, fsurf_out)
+        f_out.to_netcdf(path=wfile, mode="w", format="NETCDF3_64BIT")
+        logger.info("Successfully created file (fsurf_out) %s", wfile)
+        f_in.close()
+        f_out.close()
+
+        # write to user_nl_clm if specified
+        if self.create_user_mods:
+            with open(os.path.join(user_mods_dir, "user_nl_clm"), "a") as nl_clm:
+                line = "fsurdat = '${}'".format(os.path.join(USRDAT_DIR, fsurf_out))
+                self.write_to_file(line, nl_clm)
+
+    def create_datmdomain_at_point(self, datm_tuple: DatmFiles):
+        """
+        Create DATM domain file at a single point
+        """
```

Comment:
oh, sorry for the confusion.  Should it just be `crete_domain_at_point`

Reply:
We have a separate function `create_domain_at_point` which creates the domain (for the surface file essentially). But in NUOPC you need both a land domain file and a domain file for the atmospheric data.

Reply:
I think this is a carry over from the need for it in MCT. Do we still want to carry over being able to do both MCT and NUOPC? You wouldn't need a domain file for single-point, but you do for regional for NUOPC.

I think we should leave this in place for now to support MCT. We should then remove the MCT specific stuff at a lter point. IMHO.

Reply:
I think this is handled, so I'm resolving.

Reply:
Correct! This code exists here because it is needed for MCT cases. Maybe at some point we are going to remove this. We have one `create_datm_at_point` equivalent inside single_point_case.py  which subsets datm at a single point. But as explained by others, this creates domain files for the subsetted datm files. This is marked as resolved.